src/server/mod.rs

```diff
@@ -1 +1,279 @@
+        deserialize_from(&mut client_stream, bincode::SizeLimit::Infinite);
+    let client_name = match recv {
+        Ok(received) => {
+            match received {
+                MessageType::Login(name) => {
+                    name
+                },
+                MessageType::Quit => {
+                    println!("Client closed connection.");
+                    return
+                },
+                _ => {
+                    // unexpected packet
+
+                    "".to_string()
+                },
+            }
+        },
+        Err(_) => {
+            println!("ERROR hit me");
+            "".to_string()
+        },
+    };
+
+    // create players
+    let host = Player {
+        own_board: board.clone(),
+        op_board: Board::init(),
+        player_type: PlayerType::Human,
+        name: name,
+        capacity: Board::targets(&board),
+    };
+
+    let client = Player {
+        own_board: Board::init(),
+        op_board: Board::init(),
+        player_type: PlayerType::Human,
+        name: client_name,
+        capacity: Board::targets(&Board::init()),
+    };
+
+    // start game
+    start(host, client, client_stream);
+
+    println!("\n{}", Yellow.paint("Bye."));
+}
+
+/// Starting the game with given parameters
+fn start(mut host: Player, mut client: Player, mut stream: TcpStream) {
+    ///////////////////////////////////////////////////////////////////////////////////////////////
+    //                    Request initial board configuration from host                          //
+    ///////////////////////////////////////////////////////////////////////////////////////////////
+
+    if host.capacity == 0 {
+        net::send(
+            &mut stream,
+            MessageType::Text("Server is setting its ships, please wait :)".to_string())
+        );
+        println!("Please set your ships:");
+        match ::model::place_ships(&mut host) {
+            Ok(()) => {},
+            Err(_) => {
+                println!("Failed placing ships!");
+                net::send(&mut stream, MessageType::Quit);
+                return
+            }
+        }
+    }
+    model::print_boards(&host);
+
+    ///////////////////////////////////////////////////////////////////////////////////////////////
+    //                    Request initial board configuration from client                        //
+    ///////////////////////////////////////////////////////////////////////////////////////////////
+    net::send(&mut stream, MessageType::RequestBoard);
+
+    loop {
+        let recv: Result<types::MessageType, DeserializeError> =
+        deserialize_from(&mut stream, bincode::SizeLimit::Infinite);
+        match recv {
+            Ok(received) => {
+                match received {
+                    MessageType::Board(vec) => {
+                        client.own_board = vec.clone();
+                        client.capacity = Board::targets(&vec);
```

Comment:
It looks like this should be done with a method `Player::set_board()`. `capacity` and `own_board` are semantically connected. These invariants should be assured within methods of the player. Encapsulation principle and stuff